### ISSUE TITLE
crun updates/continuation

### ISSUE THREAD

**Issue by rudolphpienaar** [FNNDSC/projectman#27]
# <tt>crun</tt> updates

The new <tt>blockOnChild()</tt> method in various subclasses needs to be fully expanded

Priority:
- chpc
- slurm

**Comment by NicolasRannou** [comment]
is crun actually leaving in its own github project or part of chris? Would be good at some point to get it totally separated, with examples and testings!

**Comment by rudolphpienaar** [comment]
Yes -- I agree... I have a separate repo called cruntesting that has crun as well as scripts/machinery to test. Perhaps that should become the canonical reference.

**Comment by NicolasRannou** [comment]
Yes that would make sense to me. Moreover if it also contains tools for testing that is perfect.

**Comment by rudolphpienaar** [comment]
chpc sub class tested and working.

**Comment by rudolphpienaar** [comment]
Most up to date crun.py:
~rudolphpienaar/src/_common -- also updated to git.

**Comment by NicolasRannou** [comment]
Nice!

I also had some suggestions about the organization of the code and management of dependencies.
We could (should?) use bower or npm to import dependencies rather than having to copy crun.py, _common in X different locations.

Let's chat about it.

The best option I think is to use bower but I couldn't get it to work properly yet in the scripts directory so I put npm in the meanwhile. The issue I had with bower was that after installing bower with toor user on a machine, I couldn't use it properly as fnndsc. It should definitly be fixable just just didn't had time to look in depth into it.

If it makes sense, we can open a new issue to tackle the dependencies mangement in the FNNDSC projects. Maybe a good thing for an intern? :)

**Comment by NicolasRannou** [comment]
Follow up from previous comment: npm install -g bower should have worked but didn't. Might just be a umask issue.

**Comment by rudolphpienaar** [comment]
Any updates on this?

Also, with a fresh scripts checkout, how do I also checkout the _common dir?

**Comment by NicolasRannou** [comment]
yes, best is to use bower as explained here:
http://durban/wiki/index.php?title=Use_bower

But I guess this is another issue. If the blockOnchild for chpc and slurm have been implemented, we can close this issue. Makes sense?

**Comment by NicolasRannou** [comment]
@rudolphpienaar this one is completed too right?
